### ISSUE TITLE
clientbound/minecvraft:level_chunk_with_light

### ISSUE THREAD

**Issue by grndlauncher** [mircokroon/minecraft-world-downloader#674]
**When joining a server u are on the server for like 2-3 seconds and then u get that error, world downloader quickly downloads the chunks I think as it starts turning all the nearby chunks in the distance in red.**

**Log text:**

```
 Exception in thread "JavaFX Application Thread" java.lang.NullPointerException: Cannot invoke "game.data.coordinates.CoordinateDouble3D.getX()" because the return value of "game.data.entity.PlayerEntity.getPosition()" is null
	at gui.GuiMap.drawOtherPlayer(GuiMap.java:361)
	at gui.GuiMap.drawEntities(GuiMap.java:349)
	at gui.GuiMap$1.handle(GuiMap.java:112)
	at javafx.animation.AnimationTimer$AnimationTimerReceiver.lambda$handle$0(AnimationTimer.java:58)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.animation.AnimationTimer$AnimationTimerReceiver.handle(AnimationTimer.java:57)
	at com.sun.scenario.animation.AbstractPrimaryTimer.timePulseImpl(AbstractPrimaryTimer.java:356)
	at com.sun.scenario.animation.AbstractPrimaryTimer$MainLoop.run(AbstractPrimaryTimer.java:266)
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:571)
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:555)
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulseFromQueue(QuantumToolkit.java:548)
	at com.sun.javafx.tk.quantum.QuantumToolkit.lambda$runToolkit$11(QuantumToolkit.java:352)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
	at java.base/java.lang.Thread.run(Thread.java:1583)

 Exception in thread "JavaFX Application Thread" java.lang.NullPointerException
```

Screenshot:

![image](https://github.com/mircokroon/minecraft-world-downloader/assets/131962428/ba8f7937-c54a-4028-b6ab-cf19a381cca8)

**Version**
 - Game: Minecraft 1.20.6
 - Downloader: latest

**Comment by grndlauncher** [comment]
Hey man, thanks for the quick fix. But it seems to still happen.

**Error log:**

```
java.lang.ArrayIndexOutOfBoundsException: Index 100477 out of bounds for length 100477
	at packets.DataTypeProvider.readNext(DataTypeProvider.java:98)
	at packets.DataTypeProvider.readString(DataTypeProvider.java:127)
	at game.data.entity.EntityRegistry.lambda$updatePlayerAction$8(EntityRegistry.java:100)
	at util.ExceptionHandling.attempt(ExceptionHandling.java:12)
	at util.ExceptionHandling.attempt(ExceptionHandling.java:22)
	at game.data.entity.EntityRegistry.lambda$updatePlayerAction$9(EntityRegistry.java:86)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)

 java.lang.ArrayIndexOutOfBoundsException

 java.lang.ArrayIndexOutOfBoundsException

 java.lang.ArrayIndexOutOfBoundsException

```

But it did work at the beginning. I was a bit confused. I basically was able to move around just fine until one point it happened.

**Comment by mircokroon** [comment]
Looks like a different issue, ill see if I can resolve it as well

**Comment by mircokroon** [comment]
if it is what I think its a pretty minor and probably harmless error so not really a high priority, everything else should work fine

**Comment by grndlauncher** [comment]
Okay. As a note. I am using modded Minecraft (fabric with a bunch of mods). But even in Vanilla there are issues. The interesting part I just noticed is that I no longer receive the red text on the bottom left which says there is an error. Also no log option in the settings anymore.

![image](https://github.com/mircokroon/minecraft-world-downloader/assets/131962428/197d225d-b184-47ce-938b-186fd2f5c15a)


But I did get a new error it seems.

![image](https://github.com/mircokroon/minecraft-world-downloader/assets/131962428/116d8a4e-4f86-4fd8-b84b-c36141dc9c7c)

If you wanna get in contact for better direct communication, my discord is "molotovyc". I also just installed Java 22 SDK from Oracle. Although its still doesn't work, no pressure. I know you have more important stuff to do probably so still big thanks for the mod in general!
